data-azure-cosmos/src/test/java/io/micronaut/data/azure/entities/CosmosBook.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.azure.entities;
+
+import io.micronaut.data.annotation.DateUpdated;
+import io.micronaut.data.annotation.GeneratedValue;
+import io.micronaut.data.annotation.Id;
+import io.micronaut.data.annotation.MappedEntity;
+import io.micronaut.data.annotation.MappedProperty;
+import io.micronaut.data.annotation.Relation;
+import io.micronaut.data.annotation.Transient;
+import io.micronaut.data.annotation.event.PostLoad;
+import io.micronaut.data.annotation.event.PostPersist;
+import io.micronaut.data.annotation.event.PostRemove;
+import io.micronaut.data.annotation.event.PostUpdate;
+import io.micronaut.data.annotation.event.PrePersist;
+import io.micronaut.data.annotation.event.PreRemove;
+import io.micronaut.data.annotation.event.PreUpdate;
+import io.micronaut.data.cosmos.annotation.Container;
+import io.micronaut.data.cosmos.annotation.PartitionKey;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Container(throughputRequestUnits = 1000, throughputAutoScale = true, partitionKeyPath = "title")
```

Comment:
`partitionKeyPath` should be implemented as `@PartitionKey` annotation on the entity

Reply:
It is implemented if partition key is defined on the field then it would override this value. Also, we may have cases when partition key will be more complex and cannot be defined on field level simply. Something like this in Spring examples:
```
@Container(containerName = "nested-partition-key", partitionKeyPath = "/nestedEntity/nestedPartitionKey")
```
That's why I thought would be useful to have this option. And also later, if we have more than one entity per container we don't need to define on the entity but rather on the container level.

Reply:
Ok